python/lsst/daf/base/propertyContainer/propertyContainerContinued.py

```diff
@@ -489,3 +534,6 @@ def toOrderedDict(self):
         for name in self.getOrderedNames():
             d[name] = _propertyContainerGet(self, name, returnStyle=ReturnStyle.AUTO)
         return d
+
+    def __reduce__(self):
```

Comment:
I dont understand why we have a __reduce__ in this case. It seems fairly strait forward to simply define a __getstate__, and a __setstate__ on this object and let the interpreter take care of the rest, especially as we already have these function defined. I might be missing some subtle point though, if so this is fine.

Reply:
Good point! In the original version of `propertyList.cc` you'll find an extensive code comment explaining why this won't work -- it causes a segfault. I'll transfer a version of that code comment to this code to preserve that knowledge. I had meant to do that all along, but forgot.

Reply:
I know this isn't the standard Python wisdom, but I generally prefer `__reduce__` to `__getstate__`/`__setstate__` in any context; any mode of constructing an object that bypasses its constructors gives me the heebie jeebies.  Bypassing the constructor of a wrapped C++ object is a particularly dangerous thing to do, so I'm not surprised it ran in to trouble, though I don't know the details of this particular case, or even how pybind11 could have emulated bypassing the C++ constructor in making the instance to pass to `__setitem__`.

Reply:
I just made that change. I intentionally made them code comments instead of doc strings because they are internal details, not part of the API.

Reply:
@TallJimbo if you are curious look at the old `propertyList.cc`. Updating that C++ for pybind11's new pickle support improved it a little, but it was still calling python functions (because getting and restoring state is so much simpler in Python), which was quite complicated. The new Python code is a lot simpler. I transferred the existing explanation, but I confess that I do not fully understand it.